otherlibs/systhreads/st_stubs.c

```diff
@@ -336,7 +348,14 @@ static caml_thread_t caml_thread_new_info(void)
 
 /* Allocate a thread descriptor block. */
 
-static value caml_thread_new_descriptor(value clos)
+static value caml_thread_new_ident()
+{
+   value id = Val_long(thread_next_ident);
+   thread_next_ident++;
+   return id;
+}
+
+static value caml_thread_new_descriptor(value clos, caml_thread_t th)
```

Comment:
Why pass the (partially uninitialized) `caml_thread_t` instead of just the `uniq_ident` that you need? 
This is bad for maintainability.